File: test-project/greeter.provider.test/src/main/java/tests/GreeterLocator.java
package tests;

import examples.greeter.api.Greeter;

import java.util.ServiceLoader;

/**
 * Test docs
 */
public class GreeterLocator {
    /**
     * Test docs
     * @return test docs
     */
    public Greeter findGreeter() {
        return ServiceLoader.load(Greeter.class).findFirst().orElseThrow(() -> new RuntimeException("No Greeter found"));
    }
}

File: test-project-kotlin-pre-1-7/greeter.provider.test/src/main/kotlin/tests/GreeterLocator.kt
package tests

import examples.greeter.api.Greeter

import java.util.ServiceLoader

class GreeterLocator {
    fun findGreeter(): Greeter {
        return ServiceLoader.load(Greeter::class.java)
                .findFirst()
                .orElseThrow{RuntimeException("No Greeter found")}
    }
}
